core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -487,6 +487,9 @@ private void compactSingleFile(CompactionRequest compactionRequest,
             }
             long remainingBytes = fileSize - scheduledOffset;
             if (header.totalLength() > remainingBytes) {
+               // The file was modified during compaction.
+               if (handle.getFileSize() > fileSize) break;
```

Comment:
This is not quite what we want. We want to only break if the file is a logFile. Instead what I would do is add something like
```java
if (isLogFile && scheduledOffset == fileSize) {
   log.tracef("Log file %d compacted up to %d but file is now larger, ignoring remaining contents", scheduledFile, scheduledOffset);
   break;
}